README.md

```diff
@@ -1,16 +1,185 @@
+
+> "The `Array` called `array_of_arrays` has two elements. One element is an
+> `Array` containing 3 elements. Those elements are the first odd numbers greater
+> than `0`. The other element in `array_of_arrays` contains 3 elements that are
+> the first even numbers greater than 0."
+
+Is that _fun_ to read? Not really. Is it easy to think about? Not really.
+
+Now, imagine what the
+paragraph to describe agricultural output of South America looks like.  It makes
+us want to run away screaming.
+
+Humans
+learned long ago  that reading complex paragraphs like those is **not** a real strength of
+ours (Daniel Kahnemann in his book _Thinking Fast and Slow_ discusses this at
+length.).
+
+The human animal doesn't like to think methodically (like a computer). When we
+encounter complex paragraphs we skip sentences, we look ahead, etc.  We might
+even have to take notes, or track the words with a pencil to keep our focus.
+
+Humans have learned that we can share information like this far better in a bar
+chart, a line graph, or a pie chart. Economic outputs are perfect for _tables_.
+
+Unfortunately, computers don't understand pie charts. But if we structure our
+data in NDS', we can get something that's easier for humans to read **and**
+that the computer can work with.
+
+Provided you understand `Array` literal syntax, your brain loves seeing data in
+this visually "rhythmic" way.  The NDS uses space and patterns to communicate _both_ the
+content inside, as well as describe some  rules (_In this `Array` there are
+`Array`s. It seems like most inner `Array`s have three elements._).
+
+For clarity in thinking and communicating, working with NDS' is a vital skill.
+
+## Nested Data Structures Are Usable by Computers
+
+We said it before, but a computer can't work with a pie chart or a table. The
+closest approximation it can work with is a nested data structure. So NDS' make
+it _easier_ for humans to understand complex data, but they're the _only_ way a
+computer can understand complex data.
+
+## Nested Data Structures Are Transformed to Produce Insight
+
+Early (1940s-1980s) programmers pretty much had one job description:
+
+> "Take a bunch of raw data and output a re-organized version of it, or do a
+> calculation from all of it."
+
+In fact, the complex world of programming was often housed in a department
+called "data processing."
+
+Imagine you're back in that early era of "data processing." Let's now suppose
+someone asks us to:
+
+> Get the element in the `1` index for all the nested data.
+
+Thanks to the human-brain-friendly nested structure, we can look at the code
+and pretty quickly come up with `[3, 4]`. Since we know what the result should
+be thanks to "eyeballing it," coding this solution can be verified quickly:
+
+```ruby
+array_of_arrays = [ # containing Array (of Array)
+  [1, 3, 5], # "Nested" Array 1
+  [2, 4, 6]  # "Nested" Array 2
+]
+
+def find_firsts(src)
+  result = []
+  0.upto(src.count - 1) do |inner_count|
```

Comment:
In my opinion, this is fairly complex for our students at this stage. I don't remember us introducing `upto`, but also `src.count - 1` needs some explicit explanation